### PR TITLE
Cart access

### LINKED ISSUE #43: User Story 30, Visitors must register or log in to check out
```
As a visitor
When I have items in my cart
And I visit my cart
I see information telling me I must register or log in to finish the checkout process
The word "register" is a link to the registration page
The words "log in" is a link to the login page
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #49

child of #79

depends on #47

depends on #76

depends on #70

depends on #66

### PR DESCRIPTION
**Pull request by teresa-m-knowles**
As a visitor
When I have items in my cart
And I visit my cart
I see information telling me I must register or log in to finish the checkout process
The word "register" is a link to the registration page
The words "log in" is a link to the login page

Closes #43 

screen shot 2019-02-16 at 8 46 07 pm

### CODE REVIEW

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -4,8 +4,12 @@ class CartsController < ApplicationController
   before_action :require_shopper
 
   def show
-    flash[:primary] = "Your cart is empty." if @cart.total_count == 0
     set_cart
+    flash[:primary] = "Your cart is empty." if @cart.total_count == 0
+
+    if !current_user && @cart.total_count > 0
+      flash[:primary] = %Q[You must #{view_context.link_to("login", login_path)} or  #{view_context.link_to("register", register_path)} to checkout.]
```
Comment (allieIsOkay) [inline app/controllers/carts_controller.rb]: I'm not familiar with view_context, could you explain what this is doing?

**Review by allieIsOkay — APPROVED** [review]

**Review by timnallen — APPROVED** [review]
Great job! This is awesome!

**Comment by teresa-m-knowles** [comment]
Actually, just realized this broke something else. Deleting this PR for now
